2.0.0-RC2 Spring Boot + Apache HttpClient5 fails with exception stacktrace

**Describe the bug**

Reported in the chat. Application loading fails with the stacktrace below. Background is that Apache HttpClient got removed from being the request factory that Fahrschein is bound to in the spring boot starter. The Spring native dependency (which is just a wrapper), should be setting the streaming mode off, but this apparently doesn't seem to happen (see also https://github.com/spring-projects/spring-framework/issues/21650)

**Expected behavior**

- The application should load.
- Users should be made aware of this breaking change (ie. that they need to include a specific http client)

**Stacktrace**

```
org.springframework.context.ApplicationContextException: Failed to start bean 'partnerOfferableQuantityUpdatedEventNakadiListenerContainer'
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:182) ~[spring-context-6.0.12.jar:6.0.12]
	at org.springframework.context.support.DefaultLifecycleProcessor$LifecycleGroup.start(DefaultLifecycleProcessor.java:357) ~[spring-context-6.0.12.jar:6.0.12]
	at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
	at org.springframework.context.support.DefaultLifecycleProcessor.startBeans(DefaultLifecycleProcessor.java:156) ~[spring-context-6.0.12.jar:6.0.12]
	at org.springframework.context.support.DefaultLifecycleProcessor.onRefresh(DefaultLifecycleProcessor.java:124) ~[spring-context-6.0.12.jar:6.0.12]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:958) ~[spring-context-6.0.12.jar:6.0.12]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:611) ~[spring-context-6.0.12.jar:6.0.12]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.1.4.jar:3.1.4]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:737) ~[spring-boot-3.1.4.jar:3.1.4]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439) ~[spring-boot-3.1.4.jar:3.1.4]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-3.1.4.jar:3.1.4]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1309) ~[spring-boot-3.1.4.jar:3.1.4]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1298) ~[spring-boot-3.1.4.jar:3.1.4]
	at org.zalando.palladium.PalladiumApplicationKt.main(PalladiumApplication.kt:14) ~[main/:?]
Caused by: java.lang.UnsupportedOperationException: getBody not supported
	at org.springframework.http.client.HttpComponentsStreamingClientHttpRequest.getBodyInternal(HttpComponentsStreamingClientHttpRequest.java:94) ~[spring-web-6.0.12.jar:6.0.12]
	at org.springframework.http.client.AbstractClientHttpRequest.getBody(AbstractClientHttpRequest.java:60) ~[spring-web-6.0.12.jar:6.0.12]
	at org.zalando.fahrschein.http.spring.RequestAdapter.getBody(RequestAdapter.java:46) ~[fahrschein-http-spring-2.0.0-RC2.jar:?]
	at org.zalando.fahrschein.ProblemHandlingRequest.getBody(ProblemHandlingRequest.java:145) ~[fahrschein-2.0.0-RC2.jar:?]
	at org.zalando.fahrschein.NakadiClient.subscribe(NakadiClient.java:207) ~[fahrschein-2.0.0-RC2.jar:?]
	at org.zalando.fahrschein.SubscriptionBuilder.subscribe(SubscriptionBuilder.java:82) ~[fahrschein-2.0.0-RC2.jar:?]
	at org.zalando.spring.boot.fahrschein.nakadi.config.FahrscheinNakadiConsumer.getSubscription(FahrscheinNakadiConsumer.java:144) ~[fahrschein-spring-boot-starter-2.0.0-RC2.jar:?]
	at org.zalando.spring.boot.fahrschein.nakadi.config.FahrscheinNakadiConsumer.runnable(FahrscheinNakadiConsumer.java:64) ~[fahrschein-spring-boot-starter-2.0.0-RC2.jar:?]
	at org.zalando.spring.boot.fahrschein.nakadi.config.NakadiListenerContainer.start(NakadiListenerContainer.java:61) ~[fahrschein-spring-boot-starter-2.0.0-RC2.jar:?]
	at org.springframework.context.support.DefaultLifecycleProcessor.doStart(DefaultLifecycleProcessor.java:179) ~[spring-context-6.0.12.jar:6.0.12]
```

**Environment:**
 - Fahrschein Version `2.0.0-RC2`

--- a/fahrschein-spring-boot-starter/src/main/java/org/zalando/spring/boot/fahrschein/nakadi/config/RequestFactories.java
+++ b/fahrschein-spring-boot-starter/src/main/java/org/zalando/spring/boot/fahrschein/nakadi/config/RequestFactories.java
@@ -37,7 +37,7 @@ public class RequestFactories {
     private static ClientHttpRequestFactorySettings httpConfigToClientHttpRequestFactorySettings(HttpConfig httpConfig) {
         return new ClientHttpRequestFactorySettings(
                 Duration.ofMillis((int) MILLISECONDS.convert(httpConfig.getConnectTimeout().getAmount(), httpConfig.getConnectTimeout().getUnit())),
-                Duration.ofMillis((int) MILLISECONDS.convert(httpConfig.getSocketTimeout().getAmount(), httpConfig.getSocketTimeout().getUnit())), false);
+                Duration.ofMillis((int) MILLISECONDS.convert(httpConfig.getSocketTimeout().getAmount(), httpConfig.getSocketTimeout().getUnit())), true);
     }
 
 }

